verticapy/machine_learning/memmodel/decomposition.py

```diff
@@ -34,6 +34,109 @@ class PCA(InMemoryModel):
         Matrix   of   the   principal   components.
     mean: ArrayLike
         List of the averages of each input feature.
+
+    .. note:: :py:mod:`verticapy.machine_learning.memmodel` are defined entirely by their attributes.
+        For example, 'principal components' and 'mean' define a PCA model.
+
+    Examples
+    --------
+
+    **Initalization**
+
+    Import the required module.
+
+    .. ipython:: python
+            :suppress:
+
+        from verticapy.machine_learning.memmodel.decomposition import PCA
+
+    A PCA model is defined by its principal components and mean value. In this example, we will use the following:
+
+    .. ipython:: python
+            :suppress:
+
+        principal_components = [[0.4, 0.5], [0.3, 0.2]]
+        mean = [0.1, 0.3]
+
+    Let's create a :py:mod:`verticapy.machine_learning.memmodel.decomposition.PCA` model.
+
+    .. ipython:: python
+            :suppress:
+
+        model_pca = PCA(principal_components, mean)
+
+    Create a dataset.
+
+    .. ipython:: python
+            :suppress:
+
+        data = [[4, 5]]
+
+    **Making In-Memory Transformation**
+
+    Use :py:meth:`verticapy.machine_learning.memmodel.decomposition.PCA.transform` method to do transformation
+
+    .. ipython:: python
+            :suppress:
+
+        model_pca.transform(data)
+
+    **Deploy SQL Code**
+
+    Let's use the following column names:
+
+    .. ipython:: python
+            :suppress:
+
+        cnames = ['col1', 'col2']
+
+    Use :py:meth:`verticapy.machine_learning.memmodel.decomposition.PCA.transform_sql` method
+    to get the SQL code needed to deploy the model using its attributes
+
+    .. ipython:: python
+            :suppress:
+
+        model_pca.transform_sql(cnames)
+
+    **Perform an Oblimin Rotation**
+
+    Use :py:meth:`verticapy.machine_learning.memmodel.decomposition.PCA.rotate` method
+    to perform Oblimin (Varimax, Quartimax) rotation on PCA matrix
+
+    .. ipython:: python
+            :suppress:
+
+        model_pca.rotate()
+
+    .. note:: You can determine the type of rotation by adjusting value of gamma in
+        :py:meth:`verticapy.machine_learning.memmodel.decomposition.PCA.rotate` method.
+        It must be between 0.0 and 1.0.
+
+    Use gamma = 0.0, for Quartimax rotation:
+
+    .. ipython:: python
+            :suppress:
+
+        gamma = 0.0
+        model_pca.rotate(gamma)
+
+    Use gamma = 1.0, for Varimax rotation:
+
+    .. ipython:: python
+            :suppress:
+
+        gamma = 1.0
+        model_pca.rotate(gamma)
+
+    Use :py:meth:`verticapy.machine_learning.memmodel.decomposition.PCA.get_attributes` method
+    to check the attributes of the rotated model
+
+    .. ipython:: python
+            :suppress:
```

Comment:
my only question is: why using :suppress: everywhere? Will it not delete the entire cell?

Reply:
@mail4umar any idea?

Reply:
I directly rendered it and it looks fine. 

This is because the indentation of the suppress is wrong. So I think @abhsharma2 should remove it from everywhere.